pkg/leeway/build.go

```diff
@@ -1036,9 +1037,39 @@ func (p *Package) buildGo(buildctx *buildContext, wd, result string) (res *packa
 	}
 
 	var (
-		commands  [][]string
-		goCommand = "go"
+		commands      [][]string
+		isGoWorkspace bool
+		workFile      *modfile.WorkFile
 	)
+	if fc, err := os.ReadFile(filepath.Join(p.C.W.Origin, "go.work")); err == nil {
+		isGoWorkspace = true
+
+		workFile, err = modfile.ParseWork("go.work", fc, nil)
+		if err != nil {
+			return nil, xerrors.Errorf("cannot read go.work file: %w", err)
+		}
+		// we drop all use statements and later add the correct ones back in, starting with the src itself
+		for _, use := range workFile.Use {
+			err = workFile.DropUse(use.Path)
+			if err != nil {
+				return nil, err
+			}
+		}
+		workFile.AddNewUse("./", "")
+		workFile.SortBlocks()
+		workFile.Cleanup()
+		fc = modfile.Format(workFile.Syntax)
+		err = os.WriteFile(filepath.Join(wd, "go.work"), fc, 0644)
+		if err != nil {
+			return nil, err
+		}
+
+		// commands = append(commands, []string{"rm", "src/go.work"})
```

Comment:
Can we delete this line? I'm assuming it's from previous iterations of the implementation ☺️